pkg/scheduler/apis/config/v1/default_plugins.go

```diff
@@ -29,6 +29,7 @@ func getDefaultPlugins() *v1.Plugins {
 	plugins := &v1.Plugins{
 		MultiPoint: v1.PluginSet{
 			Enabled: []v1.Plugin{
+				{Name: names.SchedulingGates},
```

Comment:
is SchedulingGates a new plugin? is it behind a feature gate? if so, should this be considering whether the feature is enabled or not when including this or is the gating accomplished within the plugin itself?

put another way, if a user explicitly included the new alpha SchedulingGates plugin in their config without enabling the feature gate, would the config load and run successfully with that plugin inert, and then break if we ended up needing to drop the alpha feature and deleted the SchedulingGates plugin?

Reply:
+1, adding the plugin here should be predicated on the feature gate

Reply:
It's a new plugin and always enabled, but it _functions_ behind a feature gate.

```go
func (pl *SchedulingGates) PreEnqueue(ctx context.Context, p *v1.Pod) *framework.Status {
	if !pl.enablePodSchedulingReadiness || len(p.Spec.SchedulingGates) == 0 {
		return nil
	}
...
```

> +1, adding the plugin here should be predicated on the feature gate

Actually no, like VolumeBinding's Score plugin is always enabled, but it functions behind the feature gate:

https://github.com/kubernetes/kubernetes/blob/abcbaed0784baf5ed2382aae9705a8918f2daa18/pkg/scheduler/framework/plugins/volumebinding/volume_binding.go#L380-L391

Reply:
> Actually no, like VolumeBinding's Score plugin is always enabled, but it functions behind the feature gate:

@Huang-Wei I think this was bad precedence, it is safer to gate the addition of the plugin to avoid breaking configurations that explicitly add the plugin in case the feature doesn't make it beyond alpha for some reason.

Reply:
ok, I just updated the logic to enable ScheudlingGate when the feature gate is enabled. But we're inconsistent in VolumeBinding plugin, will submit a separate PR fixing this.

Reply:
Looks good to me, @liggitt ?

Reply:
Created an issue to track the VolumeBinding issue: https://github.com/kubernetes/kubernetes/issues/113705